rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -461,6 +465,25 @@ public ResponseEntity searchByExternalIds(@RequestParam MultiValueMap<String, St
         return new ResponseEntity<>(resources, HttpStatus.OK);
     }
 
+    @RequestMapping(value = PROJECTS_URL + "/{id}/attachmentUsage", method = RequestMethod.GET)
+    public @ResponseBody ResponseEntity<Map<String, Object>> getAttachmentUsage(@PathVariable("id") String id)
+            throws TException {
+        List<AttachmentUsage> attachmentUsages = attachmentService.getAllAttachmentUsage(id);
+        String serializedUsages = attachmentUsages.stream()
+                .map(usage -> wrapTException(() -> THRIFT_JSON_SERIALIZER.toString(usage)))
+                .collect(Collectors.joining(",", "{\"sw360:attachmentUsages\":[", "]}"));
+        GsonJsonParser parser = new GsonJsonParser();
+        Map<String, Object> attachmentUsageMap = parser.parseMap(serializedUsages);
+        List<Map> listOfAttachmentUsages = (List) attachmentUsageMap.get("sw360:attachmentUsages");
+        for (Map attachmentUsage : listOfAttachmentUsages) {
+            attachmentUsage.remove("revision");
+            attachmentUsage.remove("type");
+            return new ResponseEntity<>(attachmentUsageMap, HttpStatus.OK);
```

Comment:
Please remove this return statement from the for loop.

Reply:
removed return statement from the for loop